### DISCUSSION TITLE
OpenApi3 Spec Versioning

### DISCUSSION THREAD

**Discussion by adminy** [microsoft/typespec#4370] [Q&A] [answered]
slightly confused about versioning, it seems to be generating 3 yaml files instead of 1 file with different API paths to each version.

**Comment by timotheeguerin** [comment, marked as answer]
It indeed generate 1 OpenApi file per version representing the service at those different version. I don't think we ever really considered having it generate a single file with different endpoints like that. 

This would really only work in a pattern where you have `/v1/`, `/v2/` etc.
But feel free to file a feature request if that's something you'd like to see.

**Comment by adminy** [comment]
given that my use case is api gateway, its very simplistic in that it creates an api gateway resource per spec file. I thought that's not too efficient :) But then again, its not like one option is necessarity better than the other otherwise. I'm not sure, just wanted to hear thoughts.
